bot/seasons/evergreen/minesweeper.py

```diff
@@ -0,0 +1,198 @@
+                      ":regional_indicator_h::regional_indicator_i::regional_indicator_j:\n\n"
+        rows: typing.List[str] = []
+        for row_number, row in enumerate(board):
+            new_row = mapping[row_number + 1] + "    "
+            for cell in row:
+                new_row += mapping[cell]
+            rows.append(new_row)
+
+        discord_msg += "\n".join(rows)
+        return discord_msg
+
+    @commands.command(name="minesweeper")
+    async def minesweeper_command(self, ctx: commands.Context, bomb_chance: float = .2) -> None:
+        """Start a game of minesweeper."""
+        if ctx.author in self.games.keys():  # Player is already playing
+            msg = await ctx.send(f"{ctx.author.mention} you already have a game running")
+            await msg.delete(delay=2)
+            await ctx.message.delete(delay=2)
+            return
+
+        # Add game to list
+        board = self.generate_board(bomb_chance)
+        reveled_board = [["hidden" for _ in range(10)] for _ in range(10)]
+
+        await ctx.send(f"{ctx.author.mention} is playing minesweeper")
+        chat_msg = await ctx.send(self.format_for_discord(reveled_board))
+
+        await ctx.author.send("play by typing: `.reveal xy xy ...` or `.flag xy xy ...` \nclose the game with `.end`")
+        dm_msg = await ctx.author.send(self.format_for_discord(reveled_board))
+
+        self.games[ctx.author] = {
+            "board": board,
+            "reveled": reveled_board,
+            "dm_msg": dm_msg,
+            "chat_msg": chat_msg
+        }
+
+    @staticmethod
+    def get_cords(value1: str, value2: str) -> typing.Tuple[int, int]:
+        """Take in 2 values for the cords and turn them into numbers"""
+        if value1.isnumeric():
+            return ord(value2.lower()) - 97, int(value1) - 1
+        else:
+            return ord(value1.lower()) - 97, int(value2) - 1
+
+    async def reload_board(self, ctx: commands.Context) -> None:
+        """Update both playing boards."""
+        game = self.games[ctx.author]
+        await game["dm_msg"].delete()
+        game["dm_msg"] = await ctx.author.send(self.format_for_discord(game["reveled"]))
+        await game["chat_msg"].edit(content=self.format_for_discord(game["reveled"]))
+
+    @commands.dm_only()
+    @commands.command(name="flag")
+    async def flag_command(self, ctx: commands.Context, *cords) -> None:
+        """Place multiple flags on the board"""
+        board = self.games[ctx.author]["reveled"]
+        for cord in cords:
+            x, y = self.get_cords(cord[0], cord[1])
+            if board[y][x] == "hidden":
+                board[y][x] = "flag"
+
+        await self.reload_board(ctx)
+
+    async def lost(self, ctx: commands.Context) -> None:
+        """The player lost the game"""
+        game = self.games[ctx.author]
+        game["reveled"] = game["board"]
+        await self.reload_board(ctx)
+        await ctx.author.send(":fire: You lost :fire: ")
```

Comment:
Does it have to DM the user?

Reply:
Applies for most of the game, Unaware if this was discussed or not prior to make it

Reply:
`sends it to the channel for us to play` is what the issue said at least

Reply:
yes, it was decided that it should be played in dms.